recipes/cppcommon/all/conanfile.py

```diff
@@ -0,0 +1,97 @@
+    homepage = "https://github.com/chronoxor/CppCommon"
+    description = "C++ Common Library contains reusable components and patterns" \
+        " for error and exceptions handling, filesystem manipulations, math," \
+        " string format and encoding, shared memory, threading, time management" \
+        " and others."
+    topics = ("conan", "utils", "library")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"fPIC": [True, False],
+               "shared": [True, False]}
+    default_options = {"fPIC": True,
+                       "shared": False}
+    requires = ["fmt/7.0.3", "stduuid/1.0"]
+    generators = "cmake"
+    exports_sources = ["patches/**", "CMakeLists.txt"]
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": 10,
+            "clang": 6,
+            "gcc": 7,
+            "Visual Studio": 16,
+        }
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["CPPCOMMON_MODULE"] = "ON"
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("CppCommon-" + "d75b3447ecc89289539740a05fed618fa9cae4f6",
+            self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os == "Windows" and self.settings.arch == "x86":
+            raise ConanInvalidConfiguration("X86 Windows builds are not supported.")
+
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, "17")
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version:
+            if tools.Version(self.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration("cppcommon requires C++17, which your compiler does not support.")
+        else:
+            self.output.warn("cppcommon requires C++17. Your compiler is unknown. Assuming it supports C++17.")
+
+    def build(self):
+        patches = self.conan_data["patches"][self.version]
+        for patch in patches or []:
+            tools.patch(**patch)
+
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        include = os.path.join(self._source_subfolder, "include")
+        self.copy("*.h", src=include, dst="include", keep_path=True)
+        self.copy("*.inl", src=include, dst="include", keep_path=True)
```

Comment:
I did not check upstream yet, but it seems odd to cmake install and copy headers :thought_balloon:

Reply:
Fixed in 809da42